scripts/resetdb.sh

```diff
@@ -1,16 +1,59 @@
 #!/bin/bash
 
-readonly TRILLIAN_PATH=$(go list -f '{{.Dir}}' github.com/google/trillian)
-
-echo "Completely wipe and reset database 'test'."
-read -p "Are you sure? " -n 1 -r
-if [[ $REPLY =~ ^[Yy]$ ]]
-then
-    # User-supplied arguments must be first. This is because some flags, such
-    # as --defaults-extra-file, must be at the start.
-    mysql "$@" -u root -e 'DROP DATABASE IF EXISTS test;'
-    mysql "$@" -u root -e 'CREATE DATABASE test;'
-    mysql "$@" -u root -e "GRANT ALL ON test.* TO 'test'@'localhost' IDENTIFIED BY 'zaphod';"
-    mysql "$@" -u root -D test < ${TRILLIAN_PATH}/storage/mysql/storage.sql
-fi
-echo
+set -e
+
+usage() {
+  echo "$0 [--force] [--verbose] ..."
+  echo "accepts environment variables:"
+  echo " - DB_NAME"
+  echo " - DB_USER"
+  echo " - DB_PASSWORD"
+}
+
+collect_vars() {
+  # set unset environment variables to defaults
+  [ -z ${DB_USER+x} ] && DB_USER="root"
+  [ -z ${DB_NAME+x} ] && DB_NAME="test"
+  # format reused supplied environment variables
+  FLAGS=""
+  [ -z ${DB_PASSWORD+x} ] || FLAGS="${FLAGS} -p$DB_PASSWORD"
+
+  # handle flags
+  FORCE=false
+  VERBOSE=false
+  while [[ $# -gt 0 ]]; do
+    case "$1" in
+      --force) FORCE=true ;;
+      --verbose) VERBOSE=true ;;
+      *) FLAGS="${FLAGS} $1"
+    esac
+    shift 1
+  done
+}
+
+main() {
+  collect_vars "$@"
+
+  readonly TRILLIAN_PATH=$(go list -f '{{.Dir}}' github.com/google/trillian)
+
+  # what we're about to do
+  if [[ ${VERBOSE} = 'true' ]]
+  then
+    echo "-- using DB_USER: ${DB_USER}"
+  fi
+  echo "Warning: about to destroy and reset database '${DB_NAME}'"
+
+  [[ ${FORCE} = true ]] || read -p "Are you sure? " -n 1 -r
+
+  if [ -z ${REPLY+x} ] || [[ $REPLY =~ ^[Yy]$ ]]
+  then
+      echo "Resetting DB..."
+      mysql -u $DB_USER $FLAGS -e "DROP DATABASE IF EXISTS ${DB_NAME};"
+      mysql -u $DB_USER $FLAGS -e "CREATE DATABASE ${DB_NAME};"
+      mysql -u $DB_USER $FLAGS -e "GRANT ALL ON ${DB_NAME}.* TO '${DB_NAME}'@'localhost' IDENTIFIED BY 'zaphod';"
```

Comment:
I'm still not sure how this is supposed to work. Granting access for the user to connect only from localhost shouldn't work if the Trillian server/signer is running on a different machine, as will likely be the case on cloud infrastructure. It *may* work if $DB_NAME == "test", because some MySQL installations come with default privileges that grant unrestricted access to that database for all users.